numpy/lib/function_base.py

```diff
@@ -3670,17 +3883,90 @@ def _quantile_is_valid(q):
     return True
 
 
-def _quantile_ureduce_func(a, q, axis=None, out=None, overwrite_input=False,
-                           interpolation='linear', keepdims=False):
+def _validate_weights(a, q, axis, weights):
+    if weights is None:
+        return None
+    else:
+        a = asanyarray(a)
+        wgt = np.asanyarray(weights)
+
+        if not np.issubdtype(wgt.dtype, np.number):
+            # convert every element to np.float
+            # raises ValueError if any element fails to convert
+            wgt = np.vectorize(np.float)(wgt)
```

Comment:
`np.float` is just `float`.

Were you looking for `wgt = wgt.astype(float)`?

Why are you doing this at all? Why not allow weights to be `decimal.Decimal`?

Reply:
You're right, `wgt = wgt.astype(float)` works.

There might've been multiple reasons for this conversion, but the one reason I can think of right now is the line of code sitting right below it: `np.isnan(wgt)` raises if the type is decimal.